### ISSUE TITLE
Add analyzer for `Assert.True(a == b)`

### ISSUE THREAD

**Issue by dougbu** [xunit/xunit#1503]
Using the one-argument `Assert.True(...)` and `Assert.False(....)` methods to wrap a simple equality / inequality check is less obvious than the equivalent `Assert.Equal(...)` and `Assert.NotEqual(...)`.  (I wouldn't recommend warning when users choose to use the `userMessage` parameter.)

When the comparison involves `null`, the fixer should recommend `Assert.Null(...)` or `Assert.NotNull(...)`.

A few before / after examples
``` c#
Assert.True(attribute.ValidOn == AttributeTargets.Class);
...
Assert.Equal(AttributeTargets.Class, attribute.ValidOn);

Assert.True(d.IndexOf(item) == 0);
...
Assert.Equal(0, d.IndexOf(item));

Assert.True(value != null);
...
Assert.NotNull(value);
```

**Comment by bradwilson** [comment]
I'm not 100% sure that `Assert.Equal` is the right fix here, because `Assert.Equal` does something quite different from `==`. Instead, it should probably be `Assert.StrictEqual`.

Philosophically, I think people would rather use `Assert.Equal`, as it does a lot of heavy lifting for them, but the code fix should favor the result working 100% of the time rather than trying to encourage them to use something which might turn a passing test into a failing test. After all, they might have opted for `Assert.True(a == b)` precisely because `Assert.Equal(a, b)` gave them an inappropriate failure.

**Comment by dougbu** [comment]
@bradwilson sounds about right.

I should have been clearer the proposed analyzer shouldn't apply when the (in)equality check involved a type that e.g. overrides `operator ==`. I'd be fine if the analyzer only dealt with `bool`, `enum` and `int` and perhaps the `null` special case (for whatever type).

**Comment by bradwilson** [comment]
It actually has nothing to do with what happens when `==` is overridden; it has to do with the fact that `Assert.Equal` has an extremely complex, non-default behavior that is not the same as the default `==`; in order to get something which is equivalent to `==` (overridden or not), you must use `Assert.StrictEqual`.

Checking for overrides in not correct in this situation.

**Comment by dougbu** [comment]
> Checking for overrides in not correct in this situation.

Got it. My suggestion to focus on a few well-known types sounds better to me then.

**Comment by JamesTerwilliger** [comment]
@dougbu @bradwilson Taking a try at this one with this: https://github.com/xunit/xunit.analyzers/pull/173

**Comment by bradwilson** [comment]
Available in `1.7.0-pre.22`.
